Swift/Database.swift

```diff
@@ -118,6 +118,35 @@ public final class Database {
     }
     
     
+    /// Saves a document to the database. When write operations are executed concurrently,
+    /// and if conflicts occur, conflict handler will be called. Use the handler to directly
+    /// edit the document. Returning true, will save the document. Returning false, will cancel
+    /// the save operation.
+    ///
+    /// - Parameters:
+    ///   - document: The document.
+    ///   - conflictHandler: The conflict handler closure which can be used to resolve it.
+    /// - Returns: True if successful. False on failure
+    /// - Throws: An error on a failure.
+    @discardableResult public func saveDocument(
+        _ document: MutableDocument, conflictHandler: @escaping (MutableDocument, Document?) -> Bool
+        ) throws -> Bool {
+        do {
+            let conflictHandler = { (current: CBLMutableDocument, old: CBLDocument?) -> Bool in
+                if let oldDoc = old {
+                    return conflictHandler(MutableDocument(current), Document(oldDoc))
+                }
+                return conflictHandler(MutableDocument(current), nil)
```

Comment:
Can you simplify this? I think we can have only one return.

Reply:
I think MutableDocument(current) is not correct. It should be just document.

Reply:
`func handle(document: MutableDocument, oldDocument: Document?) -> Bool`??